js/plates-view/render-cross-section.ts

```diff
@@ -65,6 +66,57 @@ function drawMagma(ctx: CanvasRenderingContext2D, top: THREE.Vector2) {
   ctx.drawImage(magmaImg, scaleX(top.x) - magmaImg.width / 2, scaleY(top.y));
 }
 
+
+// Very simple approach to "animation". Divergent boundary magma will be clipped. Animation progress is defined
+// by number of draw calls. It only a small visual hint and it doesn't have to correlated with the real model.
+let magmaAnimationFrame = 0;
+// The more often cross-section is updated, the more steps the full animation cycle has to have.
+const animationStepsCount = 600 / UPDATE_INTERVAL.crossSection;
+
+function drawDivergentBoundaryMagma(ctx: CanvasRenderingContext2D, p1: THREE.Vector2, p2: THREE.Vector2, p3: THREE.Vector2, p4: THREE.Vector2) {
+  const tmp1 =  p1.clone().lerp(p2, 0.3);
+  const tmp2 = tmp1.clone();
+  tmp2.y = p4.y + (p1.y - p4.y) * 0.7;
+  const tmp3 = p2.clone().lerp(p3, 0.3);
+
+  const p1XScaled = scaleX(p1.x);
+  const p1YScaled = scaleY(p1.y);
+  const p3XScaled = scaleX(p3.x);
+  const p3YScaled = scaleY(p3.y);
+  
+  const clipRectHeight = Math.abs(p3YScaled - p1YScaled);
+  const clipRectWidth = Math.abs(p3XScaled - p1XScaled);
+  let animationStep = magmaAnimationFrame % animationStepsCount;
+  if (animationStep > 0.5 * animationStepsCount) {
+    animationStep = animationStepsCount - animationStep;
+  }
+  const animationProgress = Math.pow(animationStep / (animationStepsCount * 0.5), 0.5);
+  
+  ctx.save();
+
+  ctx.beginPath();
+  ctx.rect(Math.min(p1XScaled, p3XScaled), p1YScaled + (1 - animationProgress) * clipRectHeight, clipRectWidth, animationProgress * clipRectHeight);
+  ctx.clip();
+
+  const gradient = ctx.createLinearGradient(0, p1YScaled, 0, p3YScaled);
+  gradient.addColorStop(0, "#fc3c11");
```

Comment:
It might be nice to name some of these color RGB values and put them all in one place with descriptive (color) names. 

RGB values are very hard for me to interpret, so this isn't actually about what the colors represent, although, yes, where that makes sense to do, it would even be better.

Reply:
Sure, I'll start doing that incrementally when I stumble upon some hex values. Most of the colors are in colormap.ts and cross-section-colors.ts, but right, definitely not everything. I'll try to clean it up. 👍